sign/tbls/tbls_test.go

```diff
@@ -10,22 +10,39 @@ import (
 )
 
 func TestTBLS(test *testing.T) {
-	var err error
-	msg := []byte("Hello threshold Boneh-Lynn-Shacham")
+	BLSRoutine(test, []byte("Hello threshold Boneh-Lynn-Shacham"), 10)
+}
+
+func FuzzBLS(f *testing.F) {
+	f.Fuzz(func(t *testing.T, msg []byte, n int) {
+		if (n < 1) || (n > 100) {
+			t.Skip("n must be between 1 and 100")
+		}
+		if (len(msg) < 1) || (len(msg) > 1000) {
+			t.Skip("msg must have byte length between 1 and 1000")
+		}
+		BLSRoutine(t, msg, n)
+	})
+}
+
+func BLSRoutine(test *testing.T, msg []byte, n int) {
 	suite := bn256.NewSuite()
-	n := 10
-	t := n/2 + 1
+	th := n/2 + 1
+
 	secret := suite.G1().Scalar().Pick(suite.RandomStream())
-	priPoly := share.NewPriPoly(suite.G2(), t, secret, suite.RandomStream())
+	priPoly := share.NewPriPoly(suite.G2(), th, secret, suite.RandomStream())
 	pubPoly := priPoly.Commit(suite.G2().Point().Base())
 	sigShares := make([][]byte, 0)
```

Comment:
I wonder if this setup phase really needs to be repeated when fuzzing or whether fuzzing should occur on the message only, WDYT?

Reply:
I was thinking that fuzzing on the parameter n would have been more interesting than using just; I'd say that the cost of repeating these operations is not that expensive and leads some value in increased exploration. But if you feel it's relevant to fuzz only on n I can change it

Reply:
In general, fuzzing should be repeatable. Given the same "fuzzing corpus of input messages" I should get the same outputs/results in all runs.

Could we change this code so that there is no randomness outside of the fuzzer msg? 
Maybe seed the `RandomStream` using `msg`.